### DISCUSSION TITLE
[@types/passport-oauth2] `_StrategyOptionsBase.store` missing "boolean" option

### DISCUSSION THREAD

**Discussion by brownoxford** [DefinitelyTyped/DefinitelyTyped#65449] [Issues with a @types package]
The [current type for `_StrategyOptionsBase`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport-oauth2/index.d.ts#L94) allows optional `store` to be either `StateStore` instance or `undefined`.

```
    interface _StrategyOptionsBase {
        ...
        store?: StateStore | undefined;
        ...
    }
```

However, usage in `passport-oauth2/lib/strategy.js` has a [case for `options.store` being boolean](https://github.com/jaredhanson/passport-oauth2/blob/master/lib/strategy.js#L107-L108):

```
  if (options.store && typeof options.store == 'object') {
    this._stateStore = options.store;
  } else if (options.store) {
    this._stateStore = options.pkce ? new PKCEStateStore({ key: this._key }) : new StateStore({ key: this._key });
  } else if (options.state) {
    this._stateStore = options.pkce ? new PKCEStateStore({ key: this._key }) : new NonceStore({ key: this._key });
  } else {
    if (options.pkce) { throw new TypeError('OAuth2Strategy requires `state: true` option when PKCE is enabled'); }
    this._stateStore = new NullStore();
  }
```

[a comment by an automated account is not shown]

**Comment by joelklint** [comment]
👋 Nice spotted! If you want it fixed, feel free to make a PR with the suggested change 🙂

**Comment by brownoxford** [comment]
Thanks, Created #65491

**Comment by joelklint** [comment]
Hi! Since you modified a group of packages, I can not help you further. Nice initiative though 🙂
